core/src/main/scala/org/apache/spark/TaskContextImpl.scala

```diff
@@ -57,57 +68,75 @@ private[spark] class TaskContextImpl(
   // Whether the task has failed.
   @volatile private var failed: Boolean = false
```

Comment:
Yes, which is why I mentioned it as extension :-)
For failed, it is already valid to remove volatile